### DISCUSSION TITLE
randomFloat

### DISCUSSION THREAD

**Discussion by gilsmt** [remeda/remeda#950] [Suggest a Function]
Equivalent to https://lodash.com/docs/4.17.15#random
or https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_integer_between_two_values

**Comment by gustavopch** [comment]
Given there's already a `randomString()`, I'd suggest `randomInt()` and `randomFloat()` to cover both cases.

**Comment by eranhirsch** [comment]
I support adding `randomInt`, but I don't feel `randomFloat` provides enough value over `Math.prototype.random`.

Would you be able to send a PR to add it?

**Comment by gustavopch** [comment]
@eranhirsch Using `randomFloat` would be better than `Math.prototype.random` because you'd be able to specify the min and max values. It doesn't read as nice when we write the formula itself in the middle of the code where we need such thing, so a function is usually very useful to keep the code easy to read.

Ex.: random float between `x` and `y`.
😵‍💫 `x + Math.random() * (y - x)`
💯 `randomFloat(x, y)`

**Comment by cjquines** [comment]
i think `randomFloat` provides enough readability over `randomInt`, and is probably about as commonly used? so i'd see a case to add it

**Comment by pavitra-infocusp** [comment]
Why not just keep this library lightweight by using `@faker-js/faker` instead?

```ts
import { faker } from '@faker-js/faker';

const randomInt = faker.number.int();
```

**Comment by gustavopch** [comment]
> Why not just keep this library lightweight by using @faker-js/faker instead?
- In this line of reasoning, Remeda as a whole would probably not need to exist as:
  - Many functions are built in JS prototypes.
  - The rest of the functions exist in other packages in NPM.
- Given `randomString` already exists, other randomization functions should also exist for symmetry.
- Remeda can have many functions and still be lightweight if tree-shaking works properly.
- If you really need advanced stuff from `@faker-js/faker`, its 945.5kB may make sense; otherwise, if you really just need simple `randomInt` and `randomFloat`, Remeda could certainly be useful without introducing all those unnecessary bytes.
